packages/check-runtime/src/chopsticks.ts

```diff
@@ -0,0 +1,165 @@
+      "@acala-network/chopsticks-core"
+    )
+    const chain = await setup({ endpoint, block, runtimeLogLevel: 0 })
+    if (wasm) chain.head.setWasm(wasm as any)
+    await chain.api.isReady
+
+    const innerProvider = new ChopsticksProvider(chain)
+    await innerProvider.isReady
+
+    return (onMessage) => {
+      nActiveClients++
+      const subscriptions = new Set<string | number>()
+      return {
+        send: async (message: string) => {
+          const parsed = JSON.parse(message)
+
+          if (parsed.method === "chainHead_v1_follow") {
+            const subscription = await innerProvider.subscribe(
+              "chainHead_v1_followEvent",
+              parsed.method,
+              parsed.params,
+              (err, result) => {
+                if (err) {
+                  console.error(err)
+                } else
+                  onMessage(
+                    JSON.stringify({
+                      jsonrpc: "2.0",
+                      method: "chainHead_v1_followEvent",
+                      params: {
+                        subscription,
+                        result,
+                      },
+                    }),
+                  )
+              },
+            )
+            subscriptions.add(subscription)
+
+            onMessage(
+              JSON.stringify({
+                jsonrpc: "2.0",
+                id: parsed.id,
+                result: subscription,
+              }),
+            )
+          } else if (parsed.method === "chainHead_v1_unfollow") {
+            const id = parsed.params[0]
+            if (subscriptions.has(id)) {
+              subscriptions.delete(id)
+              await innerProvider.unsubscribe(
+                "chainHead_v1_followEvent",
+                "chainHead_v1_unfollow",
+                id,
+              )
+            }
+          } else {
+            onMessage(
+              JSON.stringify({
+                jsonrpc: "2.0",
+                id: parsed.id,
+                result: await innerProvider.send(parsed.method, parsed.params),
+              }),
+            )
+          }
+        },
+        disconnect: () => {
+          nActiveClients--
+          const subscriptionsCopy = [...subscriptions]
+          subscriptions.clear()
+          Promise.all(
+            [...subscriptionsCopy].map((id) =>
+              innerProvider.unsubscribe(
+                "chainHead_v1_followEvent",
+                "chainHead_v1_follow",
+                id,
+              ),
+            ),
+          )
+            .catch(noop)
+            .then(() => chain.close().then(() => innerProvider.disconnect()))
+            .then(() => (!nActiveClients ? destroyWorker() : null))
+            .catch(noop)
+        },
+      }
+    }
+  })
+
+const [encAccount] = Struct({
+  nonce: u32,
+  consumers: u32,
+  providers: u32,
+  sufficients: u32,
+  data: Struct({
+    free: u128,
+    reserved: u128,
+    frozen: u128,
+    flags: u128,
+  }),
+})
```

Comment:
if we had descriptors we could generate this with typedCodecs directly

Reply:
At the expense of loading the whole metadata, which IMO is a huge overkill for something this small